docs/specs/admin-api-get-users.md

```diff
@@ -0,0 +1,87 @@
+# Get Users in Admin API
+
+- [Get Users in Admin API](#get-users-in-admin-api)
+  - [Introduction](#introduction)
+  - [Use Cases](#use-cases)
+  - [Design](#design)
+    - [Get User by attribute](#get-user-by-attribute)
+      - [Error Response](#error-response)
+    - [Get User by login ID](#get-user-by-login-id)
+      - [Error Response](#error-response-1)
+    - [Get User by OAuth](#get-user-by-oauth)
+      - [Error Response](#error-response-2)
+
+## Introduction
+
+This document describes the feature for adding three immediately consistent get user APIs for different search criteria. Current `searchKeyword` in list user query is searching data in Elastic Search so it can only return cached result. 
+
+Auth0 reference:
+- https://auth0.com/docs/manage-users/user-search/retrieve-users-with-get-users-by-email-endpoint
+
+## Use Cases
+
+- An administrator needs to search users with real time consistency.
+- An administrator needs to export list of users data to a file.
+
+## Design
+
+### Get User by attribute
+
+`getUsersByStandardAttribute` query can search users by `attributeName` and `attributeValue` pair. Case-sensitive.
+- When there are accounts linking `create_new_user`, then an email can be shared by multiple users, thus return an array of `User`.
+- `attributeName` must be `email`, `phone_number` or `preferred_username`.
+```graphql
+type Query {
+  # attributeName must be `email`, `phone_number` or `preferred_username`.
+  getUsersByStandardAttribute(attributeName: String!, attributeValue: String!): [User!]!
+}
+```
+
+#### Error Response
+
+|Description|Name|Reason|Info|
+|---|---|---|---|
+|Invalid `attributeName` provided.|`Invalid`|`InvalidAttributeName`|-|
```

Comment:
As discussed in the standup meeting, we will use the same `reason` `GetUsersInvalidArgument` to indicate invalid input in this family of API.

You have enumerate the possible error message here.

```
attributeName must be email, phone_number or preferred_name
invalid Login ID key
invalid OAuth provider alias
```